test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -86,13 +86,13 @@ test.describe( 'Global styles variations', () => {
 
 		await expect(
 			page.locator(
-				'role=button[name="Colors background styles"i] >> data-testid=background-color-indicator'
+				'role=button[name="Background"i] >> .component-color-indicator'
```

Comment:
Why do we have to use class name selectors here? Could we use role selectors or `data-testid` instead?

Reply:
I was expecting this comment :) 

I preferred classnames for two reasons here:

 - First one, I don't like data-test-ids attributes when they linger on production code. I think we should avoid changes in production code that are only meant for testing. One potential solution would be to strip these in non test builds but not sure how to do that.
 - Second one is that the change in the structure of the component (more generic) means we can't specifically say that a particular indicator is "background" or other "text" unless we add extra props and extra complexity that I was not happy to add for test purposes.

Reply:
Let's not start a conversation here about whether `data-testid` is a best practice in tests. We already have [best practices](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/README.md#use-accessible-selectors) guidelines committed to the repo, as well as all the tooling recommendations in the community ([testing-library](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change#whats-with-the-data-testid-query), [playwright](https://playwright.dev/docs/locators#locate-by-test-id), etc), so I would expect us to follow them here :). You are more than welcome to open an issue or discussion and we can discuss further and rethink our best practices there!

> Second one is that the change in the structure of the component (more generic) means we can't specifically say that a particular indicator is "background" or other "text" unless we add extra props and extra complexity that I was not happy to add for test purposes.

I'm afraid that I don't follow, could you provide me an example code of what you mean here?

Reply:
We pass an array of colors to the indicators component and we don't specificy which color is which. So there's no way I can say which indicator is background or text here.

I don't want to go into the conversation about the "best" practices for testing, let's just say that I never agreed as Gutenberg tech lead that I have been consulted on any of that.

Reply:
I wanted to open a PR to update the data test id best practice and I don't see it mentioned anywhere. Did I miss anything? Also, it seems it's the only place where it was used. So I guess now there's nothing to change for this?

Reply:
If I'm testing the right thing, then I think there are deeper accessibility issues with this interface.

The button doesn't have any accessible label to say what color its set to, so that's maybe one of the reasons the test has to do something weird.

I tried testing in voiceover/safari to see what it's like, and it seems like you can't actually move focus into the color picker popover:

https://user-images.githubusercontent.com/677833/227924738-58352a14-5bd4-45a3-8968-53a37d0ec676.mp4

Reply:
This UI is the same UI we already had in the block inspector so I don't expect the UI to be completely inaccessible, that said a small bug is possible. I can reproduce the fact that the focus doesn't seem to jump within the popover when clicked. That said the label is there because buttons use the text content as label when there's no aria-label and I believe the text in these buttons is good enough to avoid aria-labels.

I'll look at the focus issue.

Reply:
That was a bug in the TabPanel component that should be fixed with this PR https://github.com/WordPress/gutenberg/pull/49368

Reply:
Thanks, appreciate the quick fix. I've merged that PR.

> That said the label is there because buttons use the text content as label when there's no aria-label and I believe the text in these buttons is good enough to avoid aria-labels.

Accessibility is at least a bit better now that the popover content can be focused to check what the current color is.

It doesn't necessarily need an aria-label, I was thinking more along the lines of some extra hidden text (`<VisuallyHidden>(current color: #FFF)<VisuallyHidden>`). At the moment it'd be a bit of a frustrating process to determine which colors are used for each element if a user can't see them.

Reply:
Good point about gradients 😬